beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -140,8 +140,17 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
 		// Set sync aggregate. New in Altair.
 		vs.setSyncAggregate(ctx, sBlk)
 
-		// Set execution data. New in Bellatrix.
-		if err := vs.setExecutionData(ctx, sBlk, head); err != nil {
+		// Get local and builder (if enabled) payloads. Set execution data. New in Bellatrix.
+		localPayload, err := vs.getLocalPayload(ctx, sBlk.Block(), head)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get local payload: %v", err)
+		}
+		builderPayload, err := vs.getBuilderPayload(ctx, sBlk.Block().Slot(), sBlk.Block().ProposerIndex())
+		if err != nil {
+			builderGetPayloadMissCount.Inc()
+			log.WithError(err).Error("Could not get builder payload")
+		}
+		if err := vs.setExecutionData(ctx, sBlk, localPayload, builderPayload); err != nil {
```

Comment:
Had some more thoughts, right now this function is doing a lot, It can even not set the execution data while still returning no error, I was thinking what if we turned this function into one that jsut compared the localpayload and builder payload and returned a valid payload, it will always set the execution data but outside the function.

Reply:
I guess it also needs to return an indicator whether the returned value was the localPayload or builderPayload

Reply:
> It can even not set the execution data while still returning no error

We want to return an error if local fails 

> jsut compared the localpayload and builder payload and returned a valid payload

I thought about this but I think it's a little over the top. It makes it harder to read and more bug prone